pandas/tests/reshape/test_pivot.py

```diff
@@ -301,6 +301,33 @@ def test_pivot_multi_functions(self):
         expected = concat([means, stds], keys=['mean', 'std'], axis=1)
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize('method', [True, False])
+    def test_pivot_multiple_columns_as_index(self, method):
+        # adding the test case for multiple columns as index (#21425)
+        df = DataFrame({'lev1': [1, 1, 1, 1, 2, 2, 2, 2],
+                        'lev2': [1, 1, 2, 2, 1, 1, 2, 2],
+                        'lev3': [1, 2, 1, 2, 1, 2, 1, 2],
+                        'values': [0, 1, 2, 3, 4, 5, 6, 7]})
+        data = [[0, 1], [2, 3], [4, 5], [6, 7]]
+        exp_index = pd.MultiIndex.from_product([[1, 2], [1, 2]],
+                                               names=['lev1', 'lev2'])
+        if method:
+            result = df.pivot(index=['lev1', 'lev2'],
+                              columns='lev3',
+                              values='values')
+            exp_columns = Index([1, 2], name='lev3')
+
+        else:
+            result = df.pivot(index=['lev1', 'lev2'],
+                              columns='lev3')
+            exp_columns = MultiIndex(levels=[['values'], [1, 2]],
```

Comment:
Can you use one of the more idiomatic `MultiIndex.from_` constructors here instead? Would help readability